modules/core/client/components/Map/MapStyleControl.js

```diff
@@ -0,0 +1,80 @@
+// External dependencies
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { DropdownButton, MenuItem, ButtonGroup } from 'react-bootstrap';
+
+// Internal dependencies
+import {
+  getMapboxToken,
+  MAP_STYLE_MAPBOX_STREETS,
+  MAP_STYLE_MAPBOX_SATELLITE,
+  MAP_STYLE_MAPBOX_OUTDOORS,
+  MAP_STYLE_OSM,
+} from './constants';
+import MapIcon from './MapIcon';
+
+export default function MapStyleControl({ mapStyle, setMapstyle }) {
+  const { t } = useTranslation('core');
+  const mapboxToken = getMapboxToken();
+
+  const mapboxStyleNames = {};
+  mapboxStyleNames[MAP_STYLE_MAPBOX_STREETS] = t('Streets');
+  mapboxStyleNames[MAP_STYLE_MAPBOX_SATELLITE] = t('Satellite');
+  mapboxStyleNames[MAP_STYLE_MAPBOX_OUTDOORS] = t('Outdoors');
+
+  // If it's an object, it'll have a name. Otherwise it's Mapbox URL in string presentation.
+  const selectedStyle = mapStyle?.name || mapStyle;
+
+  return (
+    <ButtonGroup aria-label={t('Map style')}>
+      <DropdownButton
+        dropup
+        noCaret
+        title={
+          <>
+            <MapIcon
+              mapboxStyle={
+                selectedStyle !== MAP_STYLE_OSM.name ? selectedStyle : false
+              }
+              mapboxToken={mapboxToken}
+            />
+            {mapStyle?.name || mapboxStyleNames[mapStyle]}
+          </>
+        }
+        id="map-style-picker"
```

Comment:
This will cause issues with multiple maps on one page since we'll have non-unique IDs, but it's a shortcoming of this version of ReactBootstrap. I'd leave for now and address in follow-ups. It doesn't break, just sub-optimal.